tests/test_uids.py

```diff
@@ -1,48 +1,53 @@
-
 import torch
 import pytest
 from types import SimpleNamespace
 from prompting.utils.uids import get_random_uids
 
 
 def make_mock_neuron(unique_coldkeys=False, unique_ips=False, vpermit_tao_limit=1000):
-
     axons = [
         SimpleNamespace(coldkey="a", ip="0.0.0.1", is_serving=True),
         SimpleNamespace(coldkey="a", ip="0.0.0.0", is_serving=True),
         SimpleNamespace(coldkey="b", ip="0.0.0.1", is_serving=True),
         SimpleNamespace(coldkey="b", ip="0.0.0.0", is_serving=True),
-        SimpleNamespace(coldkey="c", ip="0.0.0.2", is_serving=True),
+        SimpleNamespace(
+            coldkey="c", ip="0.0.0.2", is_serving=True
+        ),  # This is the validator coldkey
     ]
     metagraph = SimpleNamespace(
-        axons = axons,
-        validator_permit = torch.ones(len(axons), dtype=torch.bool),
-        S = torch.zeros(len(axons)),
-        n = torch.tensor(len(axons))
+        axons=axons,
+        validator_permit=torch.ones(len(axons), dtype=torch.bool),
+        S=torch.zeros(len(axons)),
+        n=torch.tensor(len(axons)),
     )
 
     return SimpleNamespace(
-        uid = 4,
-        config = SimpleNamespace(
-            neuron = SimpleNamespace(
-                vpermit_tao_limit = vpermit_tao_limit,
-                query_unique_coldkeys = unique_coldkeys,
-                query_unique_ips = unique_ips,
+        uid=4,
+        config=SimpleNamespace(
+            neuron=SimpleNamespace(
+                vpermit_tao_limit=vpermit_tao_limit,
+                query_unique_coldkeys=unique_coldkeys,
+                query_unique_ips=unique_ips,
             )
         ),
-        metagraph = metagraph
+        metagraph=metagraph,
     )
 
+
 @pytest.mark.parametrize(
-    "unique_coldkeys, unique_ips, k, expected_result", [
+    "unique_coldkeys, unique_ips, k, expected_result",
```

Comment:
We should also test that the uids function raises a specific warning/error under certain conditions.

1. len(uids)<k should log a bt.logging.warning
2. len(uids)=0 should raise a ValueError